### DISCUSSION TITLE
HowTo Update from 3.2.0 to 3.2.1 and 3.x.y

### DISCUSSION THREAD

**Discussion by clemsonfire** [MiczFlor/RPi-Jukebox-RFID#1959] [Q&A]
Good evening!

Please tell me how to do an update from an installed 3.2.0 to 3.2.1 and further versions!

I read it should be used "pull" - but I've no idea how to do this!

kind regards, Clemsonfire

**Comment by clemsonfire** [comment]
Please, could anybody give me a usefull hint or is version 3.x.x for developper only?

TIA!
C.

**Comment by AlvinSchiller** [comment]
Hi, the future3 documentation describes how to Update. 
If you are on "future3/main" and dont want to switch to "future3/develop" just skip the "git checkout" command.

https://rpi-jukebox-rfid.readthedocs.io/en/latest/migration.html#updating-your-jukebox-version-3

**Reply by clemsonfire** [reply]
Hi AlvinSchiller!
Thx 4 your reply!

I did it but

```
git pull
Es gibt keine Tracking-Informationen für den aktuellen Branch.
Bitte geben Sie den Branch an, welchen Sie zusammenführen möchten.
Siehe git-pull(1) für weitere Details.

    git pull <Remote-Repository> <Branch>

Wenn Sie Tracking-Informationen für diesen Branch setzen möchten, können Sie
dies tun mit:

    git branch --set-upstream-to=origin/<Branch> future3/main

pi@phoniebox:~/RPi-Jukebox-RFID $ cd src/webapp/
pi@phoniebox:~/RPi-Jukebox-RFID/src/webapp $ ./run_rebuild.sh
Total phys memory: 923 MB
Free phys memory : 686 MB
Free swap memory : 99 MB
Free total memory: 786 MB
Setting Node Options:
NODE_OPTIONS=--max-old-space-size=512
./run_rebuild.sh: Zeile 92: npm: Kommando nicht gefunden.
pi@phoniebox:~/RPi-Jukebox-RFID/src/webapp $ 
```

`
I'm sorry, but I'm a beginner and don't know, what to do.

Please, give me a hint for noobs!

TIA, Clemsonfire

**Reply by AlvinSchiller** [reply]
Hi,
how did you run the installation? With the one-liner as described [here](https://rpi-jukebox-rfid.readthedocs.io/en/latest/install.html#install-phoniebox-software)?

In which directory did you run the "git pull"? You have to be in "~/RPi-Jukebox-RFID".

**Reply by pabera** [reply]
> ./run_rebuild.sh: Zeile 92: npm: Kommando nicht gefunden.

You won't be able to switch from `future3/main` to `future3/develop` without a little more work. Background is: `future3/main` does not build the webapp, it only downloads the built version. This is to avoid installing Node and NPM to reduce overhead on your Phoniebox. `future3/develop` does not work that way, you have to build everything manually (that's on purpose, it's a development branch). This is why you get an error of missing dependencies when you run `run_rebuild.sh`.

I suggest the following: Back up the `shared` folder from your current installation (this should include all your audio files and customizations) and then reinstall `future3`. You should have the ability to choose which version to install at the beginning of the installation.

```
cd; GIT_BRANCH='future3/develop' bash <(wget -qO- https://raw.githubusercontent.com/MiczFlor/RPi-Jukebox-RFID/future3/develop/installation/install-jukebox.sh)
```

Be mindful of [this message](https://github.com/MiczFlor/RPi-Jukebox-RFID/blob/future3/develop/installation/routines/customize_options.sh#L234-L238) coming up in the installation. Once installed, copy back your backed up `shared` folder.
